bookmark.js

```diff
@@ -11,35 +12,40 @@ javascript:(function() {
   a.href = URL.createObjectURL(new Blob([`<!DOCTYPE html>
 <html>
 <head>
-  <title>Chat GPT Conversation</title>
+  <title>Chat GPT: ${title}</title>
   <meta name="generator" content="chatGPT Saving Bookmark"/>
 <style>
+body {
+    background-color: rgb(32,33,35);
+    color: rgb(236,236,241);
+    font-size: 16px;
+    font-family: Söhne,ui-sans-serif,system-ui,-apple-system,sans-serif,Helvetica Neue,Arial,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji;
+    line-height: 28px;
+    margin: -10px;
+}
+body > .w-full {
+    padding: 30px;
+}
+/* prompt */
 body > .w-full:nth-child(2n+1) {
-    background: lightgray;
+    background: rgb(52,53,65);
 }
+/* response */
 body > .w-full:nth-child(2n+2) {
-    background: darkgray;
-}
-body > .w-full {
-    padding: 10px;
-    margin: 10px;
-    border-radius: 5px;
+    background: rgb(68,70,84);
 }
 .whitespace-pre-wrap {
     white-space: pre-wrap;
 }
-p:first-child {
-  margin-top: 0;
-}
-p:last-child {
-  margin-bottom: 0;
+.flex-col {
+    max-width: 850px;
+    margin: 0px auto;
 }
 </style>
-<link rel="stylesheet"
-      href="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.7.0/build/styles/default.min.css"/>
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.7.0/build/styles/default.min.css"/>
 </head>
 <body>${template.innerHTML}</body></html>`], {type: 'text/html'}));
-  a.download = 'chatGPT.html';
+  a.download = title.replace(/[^a-z0-9]/gi, '_') + '.html';
```

Comment:
This will work only for the English language (I don't know how many but ChatGPT work for languages other than English)

Try text: "Co to jest Źdźbło albo Grzegżółka?".replace(/[^a-z0-9]/gi, '_')

I wanted to use something like this:

```javascript
const title = "Co to jest Źdźbło albo Grzegżółka?";
title.toLowerCase().replace(/\s+/g, '-').normalize('NFKD').replace(/[^\w]/g, '');
```

But it turns out that not all characters work with `normalize`. It deletes the Polish character "ł". So I would just use them as is. Only Replace spaces with dashes:

```javascript
title.toLowerCase().replace(/\s+/g, '-')
```
This way it will even work for Chinese, Korean, or Japanese.

Reply:
http://unicode.org/reports/tr44/#GC_Values_Table
\p{L}
\p{N}
letters and numbers

[^\p{L}\p{N}] should match on everything but letters and numbers, It depends what you are looking to do with it
you could first match spaces and replace with underscores and then match all other characters

```
const reg = /[^\s\p{L}\p{N}]/gu;
const title = "Co to jest Źdźbło albo Grzegżółka?이것은 시험이 ¿¡⸘ 这也是一个考验!8*"
title.replaceAll(reg, "-").replaceAll(/\s/g, "_")
```

should give you:
`"Co_to_jest_Źdźbło_albo_Grzegżółka-이것은_시험이_---_这也是一个考验-8-"`

It's to you how you do it, you can matching upcase for example with a Lu:
https://javascript.info/regexp-unicode <-- this one is a lighter read.

In the example you gave:
```
.normalize('NFKD')
```
doesn't decompose or delete the ł, its the 
```
[^\w]/g
```

Reply:
Based on this, I would use:
```javascript
const reg = /[^\p{L}\p{N}]+/gu;
const title = "Co to jest Źdźbło albo Grzegżółka?이것은 시험이 ¿¡⸘ 这也是一个考验!8*"
title.toLowerCase().replace(reg, "-").replace(/(^-)|(-$)/g, '');
// 'co-to-jest-źdźbło-albo-grzegżółka-이것은-시험이-这也是一个考验-8'
```

Reply:
You don't need -+ because the first regex has `+` and it will replace a sequence with a single dash.

Reply:
Ah ye I see now, I was originally thinking they may want to preserve the number of characters replaced and didn't notice your + in the first line, I assumed you would only want to remove leading and trailing but yours will produce cleaner file names.

Disregard my last post.

Reply:
You have a problem. You decide to solve it with a regex. Now you have two problems...

Reply:
Addressed in latest commit